### PR TITLE
This adds the deprecation notices for ko:// and nonroot

### PR DESCRIPTION
**Pull request by mattmoor**
Related: https://github.com/google/ko/issues/158
Related: https://github.com/google/ko/issues/160

### CODE REVIEW

**Comment by mattmoor** [comment]
cc @dprotaso @vaikas @n3wscott

**Comment by n3wscott** [comment]
Oh man I have a bad idea... What if kos:// was how you ask for default non-root images??

**Comment by mattmoor** [comment]
That is also a breaking change, `ko://` isn't new.

File: README.md [hunk 1]
```diff
@@ -35,8 +35,8 @@ paths like `github.com/google/ko/cmd`.
 
 **One of the goals of `ko` is to make containers invisible infrastructure.**
 Simply replace image references in your Kubernetes yaml with the import path for
-your Go binary, and `ko` will handle containerizing and publishing that
-container image as needed.
+your Go binary prefixed with `ko://`, and `ko` will handle containerizing and
```
Comment (ImJasonH) [inline README.md:38]: Worth taking this opportunity to add an example import path? (`ko://github.com/user/repo/cmd/server`)

Comment (mattmoor) [reply README.md]: Done.

**Comment by mattmoor** [comment]
@ImJasonH how have you and @jonjohnsonjr been managing releases?

**Comment by ImJasonH** [comment]
Jon set those up, but I believe you just create a new tag and [Actions takes care of the rest](https://github.com/google/ko/blob/master/.github/workflows/release.yml).

**Pull request merged**